robottelo/cli/factory.py

```diff
@@ -2531,3 +2536,215 @@ def setup_org_for_a_rh_repo(options=None):
         u'organization-id': org_id,
         u'repository-id': rhel_repo['id'],
     }
+
+
+def configure_env_for_provision(org=None, loc=None):
+    """
+    :param org: Default Organization that should be used in both host
+        discovering and host provisioning procedures
+    :param loc: Default Location that should be used in both host
+        discovering and host provisioning procedures
+    :return: List of created entities that can be re-used further in
+        provisioning or validation procedure (e.g. hostgroup or subnet)
+    """
+    # Create new organization and location in case they were not passed
+    if org is None:
+        org = make_org()
+    if loc is None:
+        loc = make_location()
+
+    # Create a new Lifecycle environment
+    lce = make_lifecycle_environment({'organization-id': org['id']})
+
+    # Create a Product, Repository for custom content and sync it
+    new_product = make_product({u'organization-id': org['id']})
+    new_repo = make_repository({
+        u'product-id': new_product['id'],
+        u'url': settings.rhel7_os
+    })
+    Repository.synchronize({'id': new_repo['id']})
+
+    # Content View should be promoted to be used with LC Env
+    cv = make_content_view({'organization-id': org['id']})
+    ContentView.add_repository({
+        u'id': cv['id'],
+        u'organization-id': org['id'],
+        u'repository-id': new_repo['id'],
+    })
+    ContentView.publish({'id': cv['id']})
+    cv = ContentView.info({'id': cv['id']})
+    ContentView.version_promote({
+        'id': cv['versions'][0]['id'],
+        'to-lifecycle-environment-id': lce['id'],
+    })
+
+    # Create puppet environment and associate organization and location
+    env = make_environment({
+        'location-ids': loc['id'],
+        'organization-ids': org['id'],
+    })
+
+    # Search for SmartProxy, and associate location
+    puppet_proxy = Proxy.list()[0]
+    Proxy.update({
+        u'id': puppet_proxy['id'],
+        u'location-ids': loc['id'],
+    })
+
+    # Network
+    # Search for existing domain or create new otherwise. Associate org,
+    # location and dns to it
+    _, _, domain_name = settings.server.hostname.partition('.')
+    domain = Domain.list({'search': 'name={0}'.format(domain_name)})
+    if len(domain) == 1:
+        domain = domain[0]
+        Domain.update({
+            'name': domain_name,
+            'location-ids': loc['id'],
+            'organization-ids': org['id'],
+            'dns-id': puppet_proxy['id'],
+        })
```

Comment:
I don't think anything would break here because we are associating all valid entities(nothing fake here). That's the reason we are first searching them and then associating them.